### PR TITLE
Add asset info to assets.

### PR DESCRIPTION
**Pull request by aliXsed**
* This info should be passed in when creating a new asset.
* Asset info should be updatable through an API.
* For the previously created assets, consider a default asset info.
* Check permissions before updating asset info.

### CODE REVIEW

File: frame/generic-asset/src/lib.rs [hunk 1]
```diff
@@ -209,6 +209,28 @@ impl<T: Trait> Subtrait for T {
 	type AssetId = T::AssetId;
 }
 
+/// Asset Metadata
+#[derive(Clone, PartialEq, Eq, Encode, Decode, Debug)]
+pub struct AssetInfo {
+    symbol: String,
+    decimal_points: u8,
```
Comment (jordy25519) [inline frame/generic-asset/src/lib.rs]: `decimal_places` or `decimals` only 1 point :P

File: frame/generic-asset/src/lib.rs [hunk 2]
```diff
@@ -526,6 +582,9 @@ decl_storage! {
 
 		/// The identity of the asset which is the one that is designated for paying the chain's transaction fee.
 		pub SpendingAssetId get(fn spending_asset_id) config(): T::AssetId;
+
+		/// The info for assets
+		pub AssetInfos: map hasher(twox_64_concat) T::AssetId => AssetInfo;
```
Comment (jordy25519) [inline frame/generic-asset/src/lib.rs]: nit: maybe something like: `AssetProperties` / `AssetMeta` / `AssetData`

File: frame/generic-asset/src/lib.rs [hunk 3]
```diff
@@ -209,6 +209,28 @@ impl<T: Trait> Subtrait for T {
...
 
+/// Asset Metadata
+#[derive(Clone, PartialEq, Eq, Encode, Decode, Debug)]
+pub struct AssetInfo {
+    symbol: String,
+    decimal_points: u8,
+}
+impl AssetInfo {
+	/// Create a new asset info by specifying its name/symbol and the designated number of
+	/// decimal points for the float representation of the balances
```
Comment (jordy25519) [inline frame/generic-asset/src/lib.rs]: smth like: "the number of decimal places in the asset's balance. i.e. balance x 10 ^ -`decimals` will be the value for display."

File: frame/generic-asset/src/tests.rs [hunk 4]
```diff
@@ -1196,3 +1196,95 @@ fn total_issuance_should_update_after_negative_imbalance_dropped() {
 			assert_eq!(GenericAsset::total_issuance(&asset_id), balance - 100);
 	});
 }
+
+#[test]
+fn asset_infos() {
+    ExtBuilder::default()
```
Comment (jordy25519) [inline frame/generic-asset/src/tests.rs]: Please split the unit tests by public function atleast e.g. `create()` and `update_asset_info()` should be separate

**Review by KarishmaBothara — APPROVED** [review]

File: frame/generic-asset/src/lib.rs [hunk 5]
```diff
@@ -948,13 +1011,25 @@ impl<T: Trait> Module<T> {
 		locks.retain(|l| l.id != id);
 		<Locks<T>>::insert(who, locks);
 	}
+
```
Comment (jordy25519) [inline frame/generic-asset/src/lib.rs]: Doc comment / do we need this function if all assets are created with an `AssetInfo` 💭 (can always update storage once-off on CENNZnet for those that were created prior)

File: frame/generic-asset/src/lib.rs (hunk 2 above)
Comment (jordy25519) [reply frame/generic-asset/src/lib.rs]: also should be `config()`

File: frame/generic-asset/src/lib.rs [hunk 6]
```diff
@@ -209,6 +209,28 @@ impl<T: Trait> Subtrait for T {
...
+	/// decimal points for the float representation of the balances
+	pub fn new(symbol: String, decimal_points: u8) -> Self {
+		Self { symbol, decimal_points, }
+	}
+}
+impl Default for AssetInfo {
+    fn default() -> Self {
+        Self {
+            symbol: String::default(),
+            decimal_points: 4,
```
Comment (jordy25519) [inline frame/generic-asset/src/lib.rs]: maybe a runtime constant, since `4` is a CENNZnet decision

**Review by jordy25519 — COMMENTED** [review]
lgtm overall!
some small naming nits and changes around configurability of the module

File: frame/generic-asset/src/lib.rs (hunk 1 above)
Comment (aliXsed) [reply frame/generic-asset/src/lib.rs]: Yeah :)

File: frame/generic-asset/src/lib.rs (hunk 5 above)
Comment (alexsednz) [reply frame/generic-asset/src/lib.rs:1014]: I suggest we keep it. This gives the permission only to the owner of the asset and relieves us from adding temporary code or unnecessarily use Sudo. I will add doc comment.

File: frame/generic-asset/src/lib.rs (hunk 6 above)
Comment (aliXsed) [reply frame/generic-asset/src/lib.rs]: Anyway, here we need to have a default value in Plug, independent of the cennznet. As you said we have had the hardcoded assumption of 4, it makes everything to behave as before unless there's a specific configuration.

File: frame/generic-asset/src/lib.rs (hunk 5 above)
Comment (aliXsed) [reply frame/generic-asset/src/lib.rs]: Removed fn asset_info()

**Review by jordy25519 — APPROVED** [review]
lgtm!

**Pull request merged**